custom_components/petlibro/binary_sensor.py

```diff
@@ -0,0 +1,106 @@
+"""Support for PETLIBRO binary sensors."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from logging import getLogger
+from collections.abc import Callable
+from datetime import datetime
+from functools import cached_property
+from typing import Any, cast
+from typing import Optional
+
+from homeassistant.components.binary_sensor import BinarySensorEntity, BinarySensorEntityDescription, BinarySensorDeviceClass
+from homeassistant.const import UnitOfMass, UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .devices import Device
+from .devices.feeders.feeder import Feeder
+from .devices.feeders.granary_feeder import GranaryFeeder
+from .devices.feeders.one_rfid_smart_feeder import OneRFIDSmartFeeder
+from . import PetLibroHubConfigEntry
+from .entity import PetLibroEntity, _DeviceT, PetLibroEntityDescription
```

Comment:
Cleanup unused imports

```suggestion
from __future__ import annotations

from dataclasses import dataclass
from collections.abc import Callable
from functools import cached_property
from typing import Optional

from homeassistant.components.binary_sensor import BinarySensorEntity, BinarySensorEntityDescription, BinarySensorDeviceClass
from homeassistant.core import HomeAssistant
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from .devices import Device
from .devices.feeders.granary_feeder import GranaryFeeder
from .devices.feeders.one_rfid_smart_feeder import OneRFIDSmartFeeder
from .entity import PetLibroEntity, _DeviceT, PetLibroEntityDescription
from . import PetLibroHubConfigEntry
```